src/main/java/com/cloudbees/plugins/credentials/CredentialsStoreAction.java

```diff
@@ -196,8 +197,9 @@ public HttpResponse doCreateDomain(StaplerRequest req) throws ServletException,
         getStore().checkPermission(MANAGE_DOMAINS);
         JSONObject data = req.getSubmittedForm();
         Domain domain = req.bindJSON(Domain.class, data);
-        if (getStore().addDomain(domain)) {
-            return HttpResponses.redirectTo("./domain/" + Util.rawEncode(domain.getName()));
+        String domainName = domain.getName();
+        if (domainName != null && getStore().addDomain(domain)) {
```

Comment:
@halkeye Ok, I have changed this in two places:
  
`doCreateDomain`

This is called when creating a new domain and in Jenkins this is what you get when you try to create one without a name:

<img width="825" alt="screen shot 2016-04-19 at 10 01 13" src="https://cloud.githubusercontent.com/assets/10669883/14631930/28ab9506-0616-11e6-957d-276187d6cdbd.png">

As you can see, a warning is displayed and the `Ok` button is not clickable. So, in this case, this change is only avoiding a NPE in case the button was clickable somehow.

`doConfigSubmit`

This is called when the domain is modified. If we create a domain with a name, and then modify it to remove the name this is what we get:

<img width="819" alt="screen shot 2016-04-19 at 10 01 37" src="https://cloud.githubusercontent.com/assets/10669883/14631975/5f4c2ecc-0616-11e6-88aa-aff8be1168e2.png">

The warning is displayed but the `Save` button is clickable. If it is clicked, we get the NPE I mention above and then the domain is deleted.

<img width="1020" alt="screen shot 2016-04-19 at 10 01 50" src="https://cloud.githubusercontent.com/assets/10669883/14632003/8bd35808-0616-11e6-8f3e-813577bcf35d.png">

The second scenario (`Save` button being clickable) should be fixed but it's not related to this PR I think.

So, I believe this is not changing the behaviour, just avoiding potential NPEs. WDYT?